public/app/features/templating/formatRegistry.ts

```diff
@@ -0,0 +1,212 @@
+import kbn from 'app/core/utils/kbn';
+import { Registry, RegistryItem, VariableModel, textUtil, dateTime } from '@grafana/data';
+import { map, isArray, replace } from 'lodash';
+
+export interface FormatRegistryItem extends RegistryItem {
+  formatter(value: any, args: string[], variable: VariableModel): string;
+}
+
+export const formatRegistry = new Registry<FormatRegistryItem>(() => {
```

Comment:
This looks great and is a HUGE improvement over no registry (with no metadata)

Should the namespace include template? [valueFormats](https://github.com/grafana/grafana/tree/master/packages/grafana-data/src/valueFormats) are often referred to as formats/formatter so may changing this to `templateFormatregistry` would help?

Reply:
maybe variableFormatRegistry? It's not part of any @grafana/data namespace yet (only an internal app thing) so easy to change

Reply:
or longer: interpolationFormatRegistry

Reply:
`variableFormatRegistry` is ok -- I'm not sure how well communicated variables vs template variables vs "templating" is.  For sure gets confusing when compared to macros often used in backend

But I agree this is not an urgent name change as it is internal

Reply:
Great stuff with the registry!